spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/DefaultPubSubAcknowledger.java

```diff
@@ -27,23 +27,35 @@
  * ack IDs and subscription name.
  *
  * @author João André Martins
+ * @author Doug Hoard
  */
 class DefaultPubSubAcknowledger implements PubSubAcknowledger {
 
 	private SubscriberStub subscriberStub;
 
+	private boolean asynchronousAcknowledgement;
+
 	DefaultPubSubAcknowledger(SubscriberStub subscriberStub) {
 		this.subscriberStub = subscriberStub;
 	}
 
+	public void setAsynchronousAcknowledgement(boolean asynchronousAcknowledgement) {
```

Comment:
Why just an `setAsync(boolean)` with good JavaDocs doesn't work for you?
We are clearly in the `DefaultPubSubAcknowledger` context, so no reason to overhead with its properties.